node/runtime/pangolin/src/pallets/bridge_dispatch.rs

```diff
@@ -3,18 +3,67 @@ pub use pallet_bridge_dispatch::{
 };
 
 // --- paritytech ---
-use frame_support::traits::Everything;
+use frame_support::{traits::Everything, weights::PostDispatchInfo};
+use sp_runtime::transaction_validity::TransactionValidityError;
 // --- darwinia-network ---
 use crate::*;
 use bp_messages::{LaneId, MessageNonce};
-use pallet_bridge_dispatch::Config;
+use darwinia_support::evm::DeriveEthereumAddress;
+use pallet_bridge_dispatch::{Config, EthereumCallDispatch};
+
+pub struct EthereumCallDispatcher;
+impl EthereumCallDispatch for EthereumCallDispatcher {
+	type AccountId = bp_pangolin::AccountId;
+	type Call = Call;
+
+	fn dispatch(
+		c: &Call,
+		origin: &bp_pangolin::AccountId,
+	) -> Result<
+		Option<sp_runtime::DispatchResultWithInfo<PostDispatchInfo>>,
+		TransactionValidityError,
+	> {
+		match c {
+			call @ Call::Ethereum(darwinia_ethereum::Call::transact { transaction: tx }) => {
+				let derive_eth_address = origin.derive_ethereum_address();
+				if let Err(validate_err) =
+					Ethereum::validate_transaction_in_block(derive_eth_address, tx)
```

Comment:
why not put it in the `transact`

Reply:
For normal RPC transact requests, they already checked in the pool and block. If we put this inside the `transact`, it duplicates the validation process.